### PR TITLE
Feature updates

### PR DESCRIPTION
**Pull request by VarunSAthreya**
-   Features added:
    -   To check if the token is expired or not
    -   Get the expiry date
-   Added examples and test cases
-   Improved description and readme and changelog

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,69 @@
 # jwt_decode
 
-A easy library to decode Jsonwebtokens with a single function call.
+A easy dart package to decode jsonwebtokens, and to chack and give its expiry dates.
```
Comment (sidsbrmnn) [inline README.md]: Rename jsonwebtokens as JSON Web Tokens and check spelling of "chack"

File: README.md [hunk 2]
```diff
@@ -1,15 +1,69 @@
...
+#### Getting expiry date
+
+To get the expiery date of the given token pass the token in the function `getExpireyDate()`.
+
+```dart
+import 'package:jwt_decode/jwt_decode.dart';
+
+String token = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJKV1QgRGVjb2RlIiwiaWF0IjoxNjA4NTgxNzczLCJleHAiOjE2NDAxMTc3NzMsImF1ZCI6Ind3dy5qd3RkZWNvZGUuY29tIiwic3ViIjoiQSBzYW1wbGUgSldUIiwibmFtZSI6IlZhcnVuIFMgQXRocmV5YSIsImVtYWlsIjoidmFydW4uc2F0aHJleWFAZ21haWwuY29tIiwicm9sZSI6IkRldmVsb3BlciJ9.vXE9ogUeMMsOTz2XQYHxE2hihVKyyxrhi_qfhJXamPQ';
+
+DateTime expiryDate = Jwt.getExpireyDate(token);
```
Comment (sidsbrmnn) [inline README.md]: getExpiryDate spelling

File: README.md [hunk 3]
```diff
@@ -1,15 +1,69 @@
...
+// expiryDate = 2021-12-22 01:46:13.000
+```
+
+#### Checking weather the token has expired or not
+
+To check weather the token has expired or not pass the token in the function `isExpired()`.
+
+#### Getting expiry date
+
+To get the expiery date of the given token pass the token in the function `getExpireyDate()`.
```
Comment (sidsbrmnn) [inline README.md]: Expiry date spelling

File: example/example.dart [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  // To decode the token
+  Map<String, dynamic> payload = Jwt.parseJwt(token);
+  // Print the payload
+  print(payload);
+
+  // Print one of its property(example: email):
+  print(payload['email']);
+
+//   To get expiry date
+  DateTime expiryDate = Jwt.getExpireyDate(token);
```
Comment (sidsbrmnn) [inline example/example.dart]: Spelling again

File: lib/jwt_decode.dart [hunk 5]
```diff
@@ -36,4 +36,22 @@ class Jwt {
...
+  static bool isExpired(String /*!*/ token) {
+    final DateTime expirationDate = getExpireyDate(token);
+    if (expirationDate != null) {
+      return DateTime.now().isAfter(expirationDate);
+    } else {
+      return false;
+    }
+  }
+
+  static DateTime getExpireyDate(String /*!*/ token) {
```
Comment (sidsbrmnn) [inline lib/jwt_decode.dart]: Spelling again

File: pubspec.yaml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 name: jwt_decode
-description: A Dart package to decode jsonwebtokens
+description: A Dart package to decode jsonwebtokens, and to chack and give its expiry dates
```
Comment (sidsbrmnn) [inline pubspec.yaml]: Changes to description same as README

File: test/jwt_decode_test.dart [hunk 7]
```diff
@@ -2,14 +2,34 @@ import 'package:flutter_test/flutter_test.dart';
...
+  };
 
   test('decodes a jsonwebtoken', () {
     expect(Jwt.parseJwt(token), payload);
     expect(() => Jwt.parseJwt(invalidToken), throwsFormatException);
   });
+
+  test('getting expiry date', () {
+    expect(
+        Jwt.getExpireyDate(token), DateTime.parse('2021-12-22 01:46:13.000'));
```
Comment (sidsbrmnn) [inline test/jwt_decode_test.dart]: Spelling of Expiry

File: test/jwt_decode_test.dart [hunk 8]
```diff
@@ -2,14 +2,34 @@ import 'package:flutter_test/flutter_test.dart';
...
+    expect(() => Jwt.getExpireyDate(invalidToken), throwsFormatException);
```
Comment (sidsbrmnn) [inline test/jwt_decode_test.dart]: Spelling of expiry

**Review by sidsbrmnn — CHANGES_REQUESTED** [review]
Make the requested changes and I'll merge it

**Comment by VarunAthreya** [comment]
Resoled changes.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,6 +1,5 @@
 ## [0.1.1] - <Set Date>
```
Comment (sidsbrmnn) [inline CHANGELOG.md]: One last change. Since you're adding new features, update the version to 0.2. Try and use semantic versioning every where.

**Review by sidsbrmnn — CHANGES_REQUESTED** [review]

**Review by sidsbrmnn — APPROVED** [review]
Perfect xD

**Pull request merged**